library/objective-c/Envoy.mm

```diff
@@ -0,0 +1,45 @@
+#import "library/objective-c/Envoy.h"
+
+#import "library/common/main_interface.h"
+
+
+@interface Envoy ()
+@property (nonatomic, strong) NSThread *runner;
+@end
+
+@implementation Envoy
+
+@synthesize runner;
+
+- (instancetype)initWithConfig:(NSString *)config {
+  if (self = [super init]) {
+    self.runner = [[NSThread alloc] initWithTarget:self selector:@selector(run:) object:config];
+    [self.runner start];
```

Comment:
I think we need to `return self;` here?

Reply:
Actually, probably simplest to remove the `else` and simply `return self;` below

Reply:
sg